src/react/hooks/useMutation.ts

```diff
@@ -122,8 +122,8 @@ export function useMutation<
         }
       }
 
-      if (baseOptions.onError || clientOptions.onError) {
-        baseOptions.onError?.(error);
+      if (ref.current.options?.onError || clientOptions.onError) {
+        ref.current.options?.onError?.(error);
         executeOptions.onError?.(error);
```

Comment:
```suggestion
      if (ref.current.options?.onError || clientOptions.onError) {
        ref.current.options?.onError?.(error);
        clientOptions.onError?.(error);
```
@brainkim @charle692 Does it make sense to use `clientOptions` here (which is `mergeOptions(baseOptions, execOptions)`), rather than `executeOptions`? If I'm reading this right, using `executeOptions` means `baseOptions.onError` won't be called?

Reply:
We could invoke from `clientOptions` instead of `executeOptions` which would be fine since `executeOptions` would override any `onError` / `onComplete` callback from `baseOptions`.

However, if we were ever to switch the order in which those objects are passed into `mergeOptions`, that behaviour would change and we would be calling the wrong callback. 

So my preference would be to keep things as is since it's more explicit and less potential for unwanted changes. Just my 2cents though, ultimately it's up to you both.

Reply:
Ok, I tried it (9e9f9665a0b1a3757185394322175bf2c30f9442), and now some relevant-looking tests are failing. We will continue to investigate, and maybe just adjust the tests if they're overly sensitive to this detail.